src/user-guide/writing-workflows/scheduling.rst

```diff
@@ -1976,6 +1988,108 @@ A more realistic example might have several tasks on each branch. The
 ``bar``, but configured differently to avoid the failure.
 
 
+Flaky Pipelines
+^^^^^^^^^^^^^^^
+
+Another pattern for using optional outputs is to assemble chains (or pipelines)
+of tasks where the chain is terminated on task failure.
+
+Here's a simple example:
+
+.. code-block:: cylc-graph
+
+   a? => b? => c?
+
+.. digraph:: Example
+   :align: center
+
+   a -> b [label="if a:succeeded", fontcolor="green"]
+   b -> c [label="if b:succeeded", fontcolor="green"]
+
+In Python, we might write this control flow like so:
+
+.. code-block:: python
+
+   if a():
+      if b():
+         c()
+
+Sometimes we might like to have a task at the end of the chain that runs no
+matter the outcome.
+
+.. code-block:: cylc-graph
+
+   a? => b? => c?
+
+   a:fail? | b:fail? | c:finish => end
+
+Note, ``:finish`` is shorthand for ``:succeed? | :fail?``, so this can be
+re-written as:
+
+.. code-block:: cylc-graph
+
+   a? => b? => c?
+
+   a:fail? | b:fail? | (c? | c:fail?) => end
```

Comment:
Do we have a preference for past or present tense in outputs? (I thought we did, and it was past).

Reply:
I followed the pattern of the other examples in this section.